pycontrails/models/cocip/output_formats.py

```diff
@@ -0,0 +1,2003 @@
+        Spatial grid resolution for the output, [:math:`\deg`]
+    optical_depth_threshold : float
+        Sensitivity of cirrus detection, set at 0.1 to match the capability of satellites.
+    seed : int
+        A number used to initialize a pseudorandom number generator.
+
+    Returns
+    -------
+    MetDataset
+        Single-level dataset containing the high resolution natural cirrus properties.
+
+    References
+    ----------
+    - :cite:`schumannContrailCirrusPrediction2012`
+
+    Notes
+    -----
+    - The high-resolution natural cirrus coverage and optical depth is distributed randomly,
+        ensuring that the mean value is equal to the value of the original grid.
+    - Enhancing the spatial resolution is necessary because the existing spatial resolution of
+        numerical weather prediction (NWP) models are too coarse to resolve the coverage area of
+        relatively narrow contrails.
+    """
+    # Ensure the required columns are included in `met`
+    met.ensure_vars(
+        (
+            'air_temperature',
+            'specific_humidity',
+            'specific_cloud_ice_water_content',
+            'geopotential',
+            'fraction_of_cloud_cover',
+        )
+    )
+
+    # Ensure `met` only contains one time step, constraint can be relaxed in the future.
+    if len(met["time"].data) > 1:
+        raise AssertionError(
+            "`met` contains more than one time step, but function only accepts one time step. "
+        )
+
+    # Calculate tau_cirrus as observed by satellites
+    met["tau_cirrus"] = tau_cirrus(met)
+    tau_cirrus_max = met["tau_cirrus"].data.sel(level=met["level"].data[-1])
+
+    # Calculate cirrus coverage as observed by satellites, cc_max(x,y,t) = max[cc(x,y,z,t)]
+    cirrus_cover_max = met["fraction_of_cloud_cover"].data.max(dim="level")
+
+    # Increase resolution of longitude and latitude dimensions
+    lon_coords_hi_res, lat_coords_hi_res = _hi_res_grid_coordinates(
+        met["longitude"].values, met["latitude"].values, spatial_grid_res=spatial_grid_res
+    )
+
+    # Increase spatial resolution by repeating existing values (temporarily)
+    n_reps = int(
+        np.round(np.diff(met["longitude"].values)[0], decimals=2)
+        / np.round(np.diff(lon_coords_hi_res)[0], decimals=2)
+    )
+    cc_rep = _repeat_rows_and_columns(cirrus_cover_max.values, n_reps=n_reps)
+    tau_cirrus_rep = _repeat_rows_and_columns(tau_cirrus_max.values, n_reps=n_reps)
+
+    # Enhance resolution of `tau_cirrus`
+    np.random.seed(seed)
+    rand_number = np.random.uniform(0, 1, np.shape(tau_cirrus_rep))
+    dx = 0.03  # Prevent division of small values: calibrated to match the original cirrus cover
+    has_cirrus = rand_number > (1 + dx - cc_rep)
+    tau_cirrus_hi_res = np.zeros_like(tau_cirrus_rep)
+    tau_cirrus_hi_res[has_cirrus] = (tau_cirrus_rep[has_cirrus] / cc_rep[has_cirrus])
+    # TODO: Zeb, using np.where would lead to RunTimeWarning because some cc_rep is zero
```

Comment:
You could use `np.divide` and pass in a `where` parameter to avoid warning.

https://numpy.org/doc/stable/reference/generated/numpy.divide.html